include/kernel.h

```diff
@@ -3002,6 +3002,23 @@ static inline bool k_work_pending(struct k_work *work)
 	return atomic_test_bit(work->flags, K_WORK_STATE_PENDING);
 }
 
+/**
+ * @brief  Check if a delayed work item is pending.
+ *
+ * This routine indicates if the work item @a work is pending in a workqueue's
+ * queue or waiting for the delay timeout.
+ *
+ * @param work Address of delayed work item.
+ *
+ * @return true if work item is waiting for the delay to expire or pending on a
+ *         work queue, or false if it is not pending.
+ */
+static inline bool k_delayed_work_pending(struct k_delayed_work *work)
+{
+	return !z_is_inactive_timeout(&work->timeout) ||
```

Comment:
Will have to get the definition for this function here.

Reply:
Id leave a comment about this being racy, also aren't we missing the check for work being detached (work->work_q == NULL) so it matches the behavior of work_cancel.

Reply:
Can add the comment.

Checking for work->worq_q == NULL I wouldn't do, as the value of work_q is not reset after the work is finished, which is a problem with work_cancel.